app/src/main/java/org/jellyfin/mobile/bridge/ExternalPlayer.kt

```diff
@@ -196,4 +202,16 @@ class ExternalPlayer(private val fragment: WebViewFragment) : KoinComponent {
             }
         }
     }
+
+    /**
+     * To ensure that the correct activity is called.
+     */
+    private fun getComponent(packageName: String): ComponentName? {
+        return when (packageName) {
+            ExternalPlayerType.MPV_PLAYER -> ComponentName(packageName, "$packageName.MPVActivity")
```

Comment:
Reading the documentation for [ComponentName](https://developer.android.com/reference/android/content/ComponentName) makes me think it should be like this:

```suggestion
            ExternalPlayerType.MPV_PLAYER -> ComponentName(packageName, "MPVActivity")
```

First parameter is `pkg`, the package. While the second is the actual class. (so no dots)

Reply:
You still need the package name for the full class name however..

Reply:
Docs on constructor:

> cls
> String: The name of the class inside of pkg that implements the component. Can not be null. This value cannot be null.

Reply:
That only means that it's inside the package, you may still need to specify the full class name (with package).

Reply:
From [MX API - Intent definition](https://sites.google.com/site/mxvpen/api#h.p_ID_481:~:text=Class%20name%20for)

> Class name for free edition is **com.mxtech.videoplayer.ad.ActivityScreen** and pro edition is **com.mxtech.videoplayer.ActivityScreen**.

From [VLC Android Player Intents](https://wiki.videolan.org/Android_Player_Intents/#Sample_code:~:text=If%20you%20specifically%20want%20start%20VideoPlayerActivity%2C%20you%20can%20set)

>  vlcIntent.setComponent(new ComponentName("org.videolan.vlc", "**org.videolan.vlc.gui.video.VideoPlayerActivity**"));

Reply:
Right, the Android documentation was a bit unclear to me.